mlflow/__init__.py

```diff
@@ -46,22 +46,52 @@
 import mlflow.tracking as tracking  # noqa: E402
 
 # model flavors
-import mlflow.fastai as fastai  # noqa: E402
-import mlflow.gluon as gluon  # noqa: E402
-import mlflow.h2o as h2o  # noqa: E402
-import mlflow.keras as keras  # noqa: E402
-import mlflow.lightgbm as lightgbm  # noqa: E402
-import mlflow.mleap as mleap  # noqa: E402
-import mlflow.onnx as onnx  # noqa: E402
-import mlflow.pyfunc as pyfunc  # noqa: E402
-import mlflow.pytorch as pytorch  # noqa: E402
-import mlflow.sklearn as sklearn  # noqa: E402
-import mlflow.spacy as spacy  # noqa: E402
-import mlflow.spark as spark  # noqa: E402
-import mlflow.statsmodels as statsmodels  # noqa: E402
-import mlflow.tensorflow as tensorflow  # noqa: E402
-import mlflow.xgboost as xgboost  # noqa: E402
-import mlflow.shap as shap  # noqa: E402
+_model_flavors_supported = []
+try:
+    # pylint: disable=unused-import
+    import pandas  # noqa: E402
+    import numpy  # noqa: E402
+except ImportError as e:
+    # We are conditional loading these commands since the skinny client does
+    # not support them due to the pandas and numpy dependencies of MLflow Models
+    pass
+else:
```

Comment:
I'd rather we just try to import the flavor modules directly and catch `ImportError` (rather than try/catching import error on numpy and pandas preemptively). Our test coverage for the complete / core client will verify that all of these modules are successfully imported.

Reply:
done